### PR TITLE
Add 3D secure require flag to braintree plugin dynamic configuration

### PR DESCRIPTION
**Pull request by salwator**
Flag to force 3D secure on payments in braintree payment gateway implementation is hard-coded.
This PR moves this setting to plugin dynamic configuration.

Additionally, since there was no mechanism for updating plugin config after config structure is changed, I've added additional update function to _BasePlugin_.

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for bdaece4588ba9b48dcdbe0a93164f4573c0f81bc (mirumee/saleor @ move-3d-secure-config-to-plugin-braintree)
Base comparison is 27e5039423219b2a35cae0e1b7f3f07642e3691e.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         38	         38	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         52	         52	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         15	         15	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         18	         18	              8
```

</p>
</details>

File: saleor/extensions/base_plugin.py [hunk 1]
```diff
@@ -221,19 +222,38 @@ def _hide_secret_configuration_fields(cls, configuration: List[dict]):
...
+        missing_keys = desired_config_keys - configured_keys
+
+        if not missing_keys:
+            return
+
+        default_config = cls._get_default_configuration()["configuration"]
+        update_values = [copy(k) for k in default_config if k["name"] in missing_keys]
+        config.extend(update_values)
+
+        configuration.save(update_fields=["configuration"])
```
Comment (korycins) [inline saleor/extensions/base_plugin.py]: This save will not do anything in case when configuration will be `None`
```
config = configuration.configuration or []
```
`config` can be a new list. You need to assign it to `configuration` obj

File: tests/extensions/test_manager.py [hunk 2]
```diff
@@ -294,6 +299,37 @@ def test_manager_save_plugin_configuration():
...
+def test_plugin_updates_configuration_shape():
+    new_config = {"name": "Foo", "value": "bar"}
+
+    @classmethod
+    def new_default_configuration(cls):
+        defaults = {
+            "name": "Sample Plugin",
+            "description": "",
+            "active": True,
+            "configuration": [{"name": "Test", "value": True}, new_config],
```
Comment (korycins) [inline tests/extensions/test_manager.py]: you could add tests for case when configuration is `None`
